### PR TITLE
[vcpkg] Fix gcc-9 warning

### PR DESCRIPTION
**Pull request by cbezault**

### CODE REVIEW

**Review by Rastaban — APPROVED** [review]

File: toolsrc/src/vcpkg/export.cpp [hunk 1]
```diff
@@ -339,26 +339,27 @@ namespace vcpkg::Export
...
-                {
-                    for (auto&& opt : implying_opts)
-                        Checks::check_exit(VCPKG_LINE_INFO,
-                                           !maybe_lookup(options.settings, opt.name),
-                                           "%s is only valid with %s",
-                                           opt.name,
-                                           main_opt_name);
-                }
-            };
+        const auto options_implies = [&](const std::string& main_opt_name,
```
Comment (ubsan) [inline toolsrc/src/vcpkg/export.cpp]: Should this be a `StringView`?

File: toolsrc/src/vcpkg/export.cpp [hunk 2]
```diff
@@ -339,26 +339,27 @@ namespace vcpkg::Export
...
+                                         bool main_opt,
```
Comment (ubsan) [inline toolsrc/src/vcpkg/export.cpp]: could you rename this `is_main_option` or something?

**Review by strega-nil — APPROVED** [review]

File: toolsrc/src/vcpkg/export.cpp (hunk 1 above)
Comment (cbezault) [reply toolsrc/src/vcpkg/export.cpp]: Well, a `StringLiteral` but yeah pretty much

File: toolsrc/src/vcpkg/export.cpp (hunk 2 above)
Comment (cbezault) [reply toolsrc/src/vcpkg/export.cpp]: sure

**Pull request merged**